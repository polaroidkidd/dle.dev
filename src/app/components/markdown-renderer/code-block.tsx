import React from "react";
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter";
import { tomorrow } from "react-syntax-highlighter/dist/esm/styles/prism";

interface ICodeBlockProps {
  value: string;
  language: string;
}

const CodeBlock: React.FC<ICodeBlockProps> = ({value, language}) => {
  return (
    <SyntaxHighlighter
      language={language}
      style={tomorrow}
      customStyle={{fontSize: "1.5rem"}}
    >
      {value}
    </SyntaxHighlighter>
  );
};

export { CodeBlock };