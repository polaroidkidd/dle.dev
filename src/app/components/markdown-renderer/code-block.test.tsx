import React from "react";
import renderer from "react-test-renderer";
import { CodeBlock } from "./code-block";

test("Render Navigation", () => {
  const tree = renderer.create(
    <CodeBlock value="import React, { Component } from 'react';" language="typescript"/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});