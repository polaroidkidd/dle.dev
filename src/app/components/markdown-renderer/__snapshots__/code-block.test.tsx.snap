// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Navigation 1`] = `
<pre
  style={
    Object {
      "MozHyphens": "none",
      "MozTabSize": "4",
      "OTabSize": "4",
      "WebkitHyphens": "none",
      "background": "#2d2d2d",
      "color": "#ccc",
      "fontFamily": "Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace",
      "fontSize": "1.5rem",
      "hyphens": "none",
      "lineHeight": "1.5",
      "margin": ".5em 0",
      "msHyphens": "none",
      "overflow": "auto",
      "padding": "1em",
      "tabSize": "4",
      "textAlign": "left",
      "whiteSpace": "pre",
      "wordBreak": "normal",
      "wordSpacing": "normal",
      "wordWrap": "normal",
    }
  }
>
  <code
    style={
      Object {
        "MozHyphens": "none",
        "MozTabSize": "4",
        "OTabSize": "4",
        "WebkitHyphens": "none",
        "background": "none",
        "color": "#ccc",
        "fontFamily": "Consolas, Monaco, 'Andale Mono', 'Ubuntu Mono', monospace",
        "hyphens": "none",
        "lineHeight": "1.5",
        "msHyphens": "none",
        "tabSize": "4",
        "textAlign": "left",
        "whiteSpace": "pre",
        "wordBreak": "normal",
        "wordSpacing": "normal",
        "wordWrap": "normal",
      }
    }
  >
    <span
      className="token"
      style={
        Object {
          "color": "#cc99cd",
        }
      }
    >
      import
    </span>
     
    <span
      className="token maybe-class-name"
      style={Object {}}
    >
      React
    </span>
    <span
      className="token"
      style={
        Object {
          "color": "#ccc",
        }
      }
    >
      ,
    </span>
     
    <span
      className="token"
      style={
        Object {
          "color": "#ccc",
        }
      }
    >
      {
    </span>
     
    <span
      className="token maybe-class-name"
      style={Object {}}
    >
      Component
    </span>
     
    <span
      className="token"
      style={
        Object {
          "color": "#ccc",
        }
      }
    >
      }
    </span>
     
    <span
      className="token"
      style={
        Object {
          "color": "#cc99cd",
        }
      }
    >
      from
    </span>
     
    <span
      className="token"
      style={
        Object {
          "color": "#7ec699",
        }
      }
    >
      'react'
    </span>
    <span
      className="token"
      style={
        Object {
          "color": "#ccc",
        }
      }
    >
      ;
    </span>
  </code>
</pre>
`;
